src/core/public/chrome/ui/header/header.tsx

```diff
@@ -213,34 +215,71 @@ export function Header({
     />
   );
 
-  const renderNavToggle = () => (
-    <EuiHeaderSectionItemButton
-      data-test-subj="toggleNavButton"
-      aria-label={i18n.translate('core.ui.primaryNav.toggleNavAriaLabel', {
-        defaultMessage: 'Toggle primary navigation',
-      })}
-      onClick={() => setIsNavOpen(!isNavOpen)}
-      aria-expanded={isNavOpen}
-      aria-pressed={isNavOpen}
-      aria-controls={navId}
-      ref={toggleCollapsibleNavRef}
-      className={
-        useUpdatedHeader
-          ? useApplicationHeader
-            ? 'newAppTopNavExpander'
-            : 'newPageTopNavExpander'
-          : undefined
-      }
-    >
-      <EuiIcon
-        type="menu"
-        size="m"
-        title={i18n.translate('core.ui.primaryNav.menu', {
-          defaultMessage: 'Menu',
+  const renderNavToggle = () => {
+    const renderNavToggleWithExtraProps = (
+      props: EuiHeaderSectionItemButtonProps & { isSmallScreen?: boolean }
+    ) => (
+      <EuiHeaderSectionItemButton
+        data-test-subj="toggleNavButton"
+        aria-label={i18n.translate('core.ui.primaryNav.toggleNavAriaLabel', {
+          defaultMessage: 'Toggle primary navigation',
         })}
-      />
-    </EuiHeaderSectionItemButton>
-  );
+        onClick={() => setIsNavOpen(!isNavOpen)}
+        aria-expanded={isNavOpen}
+        aria-pressed={isNavOpen}
+        aria-controls={navId}
+        ref={toggleCollapsibleNavRef}
+        {...props}
+        className={classnames(
+          useUpdatedHeader
+            ? useApplicationHeader
+              ? 'newAppTopNavExpander'
+              : 'newPageTopNavExpander'
+            : undefined,
+          props.className
+        )}
+      >
+        {props.isSmallScreen ? (
+          <EuiButtonIcon
+            iconType="menu"
+            size="xs"
+            title={i18n.translate('core.ui.primaryNav.menu', {
+              defaultMessage: 'Menu',
+            })}
+            display="base"
+            color="subdued"
+            iconSize="s"
+          />
+        ) : (
+          <EuiIcon
+            type="menu"
+            size="m"
+            title={i18n.translate('core.ui.primaryNav.menu', {
+              defaultMessage: 'Menu',
+            })}
+          />
+        )}
+      </EuiHeaderSectionItemButton>
+    );
+    return (
+      <>
+        {isNavOpen ? null : (
+          <EuiHideFor sizes={['xs', 's', 'm']}>
```

Comment:
`EuiHideFor` adds a `resize` listener to window to fake responsiveness and the more we use it, the more sluggish the interface will feel. Can we move this logic to CSS and use media queries instead?

Reply:
Thanks for that, updated. But the change can not be applied to the left nav as the Modal will push the page even though it is visually hidden.